File: sdk/python/src/tinyverse/types/reputation.py
from __future__ import annotations

from dataclasses import dataclass
from typing import Optional


@dataclass
class ReputationScore:
    agent_id: str
    score: int
    breakdown: dict[str, int]
    updated_at: str
    username: Optional[str] = None


@dataclass
class ReputationReview:
    review_id: str
    reviewer: str
    subject: str
    rating: int
    transaction_ref: str
    created_at: str
    comment: Optional[str] = None
    context: Optional[str] = None
    signature: Optional[str] = None


@dataclass
class ReputationReviewCreate:
    reviewer: str
    subject: str
    rating: int
    transaction_ref: str
    comment: Optional[str] = None
    context: Optional[str] = None
    signature: Optional[str] = None


@dataclass
class Attestation:
    attestation_id: str
    agent: str
    agent_crypto_id: str
    platform: str
    handle: str
    verified_at: str
    status: str
    proof_url: Optional[str] = None
    signature: Optional[str] = None


@dataclass
class AttestationCreate:
    agent: str
    agent_crypto_id: str
    platform: str
    handle: str
    proof_url: Optional[str] = None
    signature: Optional[str] = None


@dataclass
class AttestationVerification:
    verified: bool
    status: Optional[str] = None
    verified_at: Optional[str] = None
    error: Optional[str] = None


@dataclass
class ReputationHistoryPoint:
    timestamp: str
    score: int
    breakdown: Optional[dict[str, int]] = None


@dataclass
class LeaderboardEntry:
    rank: int
    username: Optional[str] = None
    crypto_id: Optional[str] = None
    score: Optional[int] = None
    transactions: Optional[int] = None
    reviews: Optional[int] = None
    group_id: Optional[str] = None
    name: Optional[str] = None
    member_count: Optional[int] = None
    messages_sent: Optional[int] = None
    unique_recipients: Optional[int] = None
    volume_usdc: Optional[str] = None
    transaction_count: Optional[int] = None
    revenue: Optional[str] = None
    sales_count: Optional[int] = None
    average_rating: Optional[float] = None
    current_score: Optional[int] = None
    previous_score: Optional[int] = None
    delta: Optional[int] = None
    unique_counterparties: Optional[int] = None
    messages_this_period: Optional[int] = None
    is_public: Optional[bool] = None
    product_count: Optional[int] = None
    account_age: Optional[str] = None


@dataclass
class LeaderboardResponse:
    leaderboard: str
    entries: list[LeaderboardEntry]
    updated_at: str
    period: Optional[str] = None
    sort: Optional[str] = None

File: sdk/src/types/reputation.ts
export interface ReputationScore {
  agentId: string;
  username?: string;
  score: number;
  breakdown: Record<string, number>;
  updatedAt: string;
}

export interface ReputationReview {
  reviewId: string;
  reviewer: string;
  subject: string;
  rating: number;
  comment?: string;
  context?: string;
  transactionRef: string;
  signature?: string;
  createdAt: string;
}

export interface ReputationReviewCreate {
  reviewer: string;
  subject: string;
  rating: number;
  comment?: string;
  context?: string;
  transactionRef: string;
  signature?: string;
}

export interface Attestation {
  attestationId: string;
  agent: string;
  agentCryptoId: string;
  platform: string;
  handle: string;
  proofUrl?: string;
  verifiedAt: string;
  status: string;
  signature?: string;
}

export interface AttestationCreate {
  agent: string;
  agentCryptoId: string;
  platform: string;
  handle: string;
  proofUrl?: string;
  signature?: string;
}

export interface AttestationVerification {
  verified: boolean;
  status?: string;
  verifiedAt?: string;
  error?: string;
}

export interface ReputationHistoryPoint {
  timestamp: string;
  score: number;
  breakdown?: Record<string, number>;
}

export interface LeaderboardEntry {
  rank: number;
  username?: string;
  cryptoId?: string;
  score?: number;
  transactions?: number;
  reviews?: number;
  groupId?: string;
  name?: string;
  memberCount?: number;
  messagesSent?: number;
  uniqueRecipients?: number;
  volumeUSDC?: string;
  transactionCount?: number;
  revenue?: string;
  salesCount?: number;
  averageRating?: number;
  currentScore?: number;
  previousScore?: number;
  delta?: number;
  uniqueCounterparties?: number;
  messagesThisPeriod?: number;
  isPublic?: boolean;
  productCount?: number;
  accountAge?: string;
}

export interface LeaderboardResponse {
  leaderboard: string;
  period?: string;
  sort?: string;
  entries: Array<LeaderboardEntry>;
  updatedAt: string;
}
